### PR TITLE
SDI-611 Create Nomis course activity rules

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResource.kt [hunk 1]
```diff
@@ -83,9 +83,6 @@ class ActivitiesResource(private val activityService: ActivityService, private v
       ApiResponse(
         responseCode = "200",
         description = "Activity information",
-        content = [
-          Content(mediaType = "application/json", schema = Schema(implementation = UpdateActivityResponse::class))
-        ]
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResource.kt:88]: UpdateActivityResponse seems to be unused - hope its ok to delete it

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRatesService.kt [hunk 2]
```diff
@@ -29,7 +29,7 @@ class PayRatesService(
         courseActivity.prison,
         rate.incentiveLevel
       )
-        ?: throw BadDataException("IEP type ${rate.incentiveLevel} does not exist for prison ${dto.prisonId}")
+        ?: throw BadDataException("Pay rate IEP type ${rate.incentiveLevel} does not exist for prison ${dto.prisonId}")
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRatesService.kt:32]: useful to discriminate from an error in the course minimum iep level

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityServiceTest.kt [hunk 3]
```diff
@@ -174,22 +177,22 @@ class ActivityServiceTest {
...
         activityService.createActivity(createRequest)
       }
       Assertions.assertThat(thrown.message).isEqualTo("IEP type STD does not exist for prison $PRISON_ID")
     }
 
-    @Test
-    fun invalidPayBandCode() {
-      whenever(availablePrisonIepLevelRepository.findFirstByAgencyLocationAndId(any(), any())).thenReturn(null)
+    // TODO @Test
+    fun invalidPayBandIEP() {
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityServiceTest.kt:189]: Ill come back to this. I tried to add a validation test for the pay band iep but got weird problems.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleRuleService.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+@Service
+class ScheduleRuleService {
+
+  fun mapRules(dto: CreateActivityRequest, courseActivity: CourseActivity) =
+    dto.scheduleRules.map {
+      validateRequest(it)
+
+      // Prod data usually uses 1st day of the scheduleStartDate month for the 'date' part
+      val monthStart = courseActivity.scheduleStartDate!!.with(TemporalAdjusters.firstDayOfMonth())
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleRuleService.kt:20]: Won't this mean that the data in Activities and Nomis might not match? Have yo uchecked this with Activities.

Also feels uncomfortable using !! here, could we take the date from the request instead?

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**